packages/client/components/TimelineFeedList.tsx

```diff
@@ -51,11 +76,35 @@ const TimelineFeedList = (props: Props) => {
   const {viewer} = data
   const {timeline} = viewer
   const lastItem = useLoadNextOnScrollBottom(paginationRes, {}, 10)
+
+  const {freeHistory, lockedHistory} = useMemo(() => {
```

Comment:
We will lock the history of an archived team as well, so I want to show the message before the first locked meeting. I think I will simply change the message to unlock the organization instead of the team. This way we should avoid some of the pitfalls you mentioned.

Reply:
Makes sense. The confusing thing for me was when I clicked the CTA button, it took me to the billing page of an org that I had deleted. When deleting an org, we show a note saying "This can't be undone" and I don't think there's any way to unarchive an org once it's been deleted, so the user won't upgrade.

Perhaps the message should show for archived teams but only if the org hasn't been deleted?

Reply:
I adapted the changes. We now also filter for organization user to check if the viewer could upgrade at all. Good catch!